python/scripts/genetic_search.py

```diff
@@ -84,23 +82,35 @@ def persist_netlist_in_run_dir(netlist: Netlist):
         min_fingers,
         max_fingers,
         precision,
-        Netlist.create(BaseNetlistFile(File(reference_netlist_path))),
-        persist_netlist_in_run_dir,
+        Netlist.create(BaseNetlistFile(File(reference_netlist))),
         seed=seed,
     )
 
-    cost_function: CostFunction = partial(delay_cost_function, delay_idx=delay_index)
-    genetic_search = GeneticSearch(
-        candidate_generator,
-        cost_function,  # TODO: parameterize when there are more cost functions
-        max_iterations,
+    cost_function: CostFunction = partial(
+        delay_cost_function,
+        delay_idx=delay_index,
+    )
+
+    search_algorithm = GeneticSearch(
+        simulator=simulator,
+        candidate_generator=candidate_generator,
+        cost_function=cost_function,  # TODO: parameterize when there are more cost functions
+        max_iterations=max_iterations,
     )
 
     # Do the sweep
     info("Starting genetic search.")
-    best_netlist = genetic_search.search()
+    best_netlist = search_algorithm.search()
     info("Search complete")
-    info(f"Find netlist is named {best_netlist.cell_name} in {netlist_work_dir_path}")
+    best_netlist.mutate(
+        cell_name=best_netlist.base_netlist_cell_name,
```

Comment:
Could also get this name from Line 85 above if `Netlist.create(BaseNetlistFile(File(reference_netlist))) ` was pulled out to a vairable instead of adding it to `Netlist` which I think would be a bit cleaner.

Reply:
I like that, I will do that.